lib/cib/cib_attrs.c

```diff
@@ -86,10 +93,12 @@ find_nvpair_attr_delegate(cib_t * the_cib, const char *attr, const char *section
     }
 
     xpath_string = calloc(1, xpath_max);
-    offset += snprintf(xpath_string + offset, xpath_max - offset, "%.128s", get_object_path(section));
```

Comment:
It needs the memory fix, but I agree the old way was more readable. I see two alternative approaches: define the macro to just the snprintf, so it looks like:

offset += attr_snprintf(xpath_string, offset, xpath_max, "//%s", node_type);

Or define a struct with xpath_string, offset and xpath_max as members, so it looks like:

attr_snprintf(xpath_builder, "//%s", node_type);

The first appoach makes it more obvious that offset is being modified, and the second approach builds on yours to put the focus on the pattern rather than a list of variable names.